File: producer/producer.js
const kafka = require('kafka-node');

// Configuración del cliente Kafka
const client = new kafka.KafkaClient({
  kafkaHost: 'localhost:9092'
});

// Configuración del productor
const producer = new kafka.Producer(client);

// Manejo del evento 'ready' para enviar mensajes
producer.on('ready', function() {
  // Mensaje a enviar
  const message = 'Hola, mundo!';
  
  // Enviar el mensaje al topic 'test'
  producer.send([{ topic: 'test', messages: [message] }], function(err, result) {
    console.log('Mensaje enviado:', message);
  });
});

producer.on('error', function(err) {
  console.error('Error en el productor:', err);
  process.exit(1); // Sale del proceso con un código de error
});

File: producer.py
from confluent_kafka import Producer
import sys

# Configuración del productor
conf = {'bootstrap.servers': 'localhost:9092'}

# Crear el productor con la configuración dada
producer = Producer(conf)

# Función de callback para saber si un mensaje fue entregado o no
def delivery_report(err, msg):
    if err is not None:
        print(f'Mensaje falló la entrega: {err}')
    else:
        print(f'Mensaje entregado a {msg.topic()} [{msg.partition()}]')

try:
    while True:
        message = input('Introduce un mensaje para enviar (o "exit" para terminar): ')

        if message.lower() == 'exit':
            break

        # Enviar el mensaje al topic 'test'
        producer.produce('test', key='key', value=message, callback=delivery_report)

        # Esperar a que los mensajes en cola sean entregados
        producer.flush()
except KeyboardInterrupt:
    pass
finally:
    # Cerrar el productor
    producer.flush()